.github/workflows/build-and-release.yml

```
@@ -1,6 +1,9 @@
 name: build-and-release
 
-on: push
+on:
+  push:
+    branches: [master]
+  pull_request:
```

Why restrict pushes to master?

```suggestion
on: [push, pull_request]
```

---

Then CI will go through and appear twice when you create a pull request with a branch being inside of this repository: once from the `pull_request` event and once from the `push` event.

---

I see (see also https://github.com/orgs/community/discussions/26276). Running the CI twice seems okay to me since it's fairly quick anyways (takes less than two minutes). Otherwise, my priority would be on running CI in PRs, which would match your current suggestion.